script.js

```diff
@@ -1,12 +1,200 @@
 //You can edit ALL of the code here
+
+let allEpisodes;
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  root.innerText = '';
+
+  let showContainer = document.getElementById("show-container");
+  if (showContainer) {
+    showContainer.innerText = '';
+  } else {
+    showContainer = document.createElement("div");
+    showContainer.id = "show-container";
+    root.appendChild(showContainer);
+  }
+
+  const searchBox = document.createElement("input");
+  searchBox.id = "search-box";
+  searchBox.placeholder = "Search for show name";
+  searchBox.addEventListener("keyup", renderShows);
+  showContainer.appendChild(searchBox);
+
+  let showSelectorContainer = document.createElement("div");
+  showSelectorContainer.id = "show-selector-container";
+  showContainer.appendChild(showSelectorContainer);
+
+  renderShows();
 }
 
-function makePageForEpisodes(episodeList) {
+function renderShows(event) {
+  window.scrollTo(0, 0);
+  const searchTerm = event && event.target.value.toLowerCase() || "";
+
+  let show = document.getElementById("show-container");
+
+  const showSelector = document.createElement("ul");
+  getAllShows().filter(show => show.name.toLowerCase().includes(searchTerm) || show.summary.toLowerCase().includes(searchTerm)).forEach(show => {
```

Comment:
To the student I'd say:
Consider moving out this anonymous arrow function that you've passed to filter, and giving it a descriptive name.  It'll help you understand your code one week from now!

Make sure you give it a good name.  What does that arrow function do?  (Tip: although it can be *used* by filter(), the arrow function itself does NOT filter anything.)

Reply:
Pulled out a `showMatchesTerm` function :)